meson.build

```diff
@@ -321,38 +318,32 @@ if cc.has_link_argument('-Wl,--nxcompat,--no-seh,--dynamicbase')
     noexecstack = true
 endif
 
-if noexecstack
-    features += 'noexecstack'
-endif
+features += {'noexecstack': noexecstack}
 
 if not get_option('build-date')
     flags += '-DNO_BUILD_TIMESTAMPS'
-else
-    features += 'build-date'
 endif
+features += {'build-date': get_option('build-date')}
 
-if get_option('ta-leak-report')
-    features += 'ta-leak-report'
-endif
+features += {'ta-leak-report': get_option('ta-leak-report')}
 
 libdl_dep = cc.find_library('libdl', required: false)
```

Comment:
unrelated but should probably use this?
https://mesonbuild.com/Dependencies.html#dl-libdl

Reply:
We could but that requires bumping the meson version to 0.62.